pkg/api/scylla/v1alpha1/types_scylladbdatacenter.go

```diff
@@ -0,0 +1,528 @@
+	// alternatorOptions designates this cluster an Alternator cluster.
+	// +optional
+	AlternatorOptions *AlternatorOptions `json:"alternatorOptions,omitempty"`
+
+	// additionalScyllaDBArguments specify a list of arguments appended to the ScyllaDB binary during startup.
+	// When set, ScyllaDB may behave unexpectedly, and every such setup is considered unsupported.
+	// +optional
+	AdditionalScyllaDBArguments []string `json:"additionalScyllaDBArguments,omitempty"`
+
+	// developerMode determines if the cluster runs in developer-mode.
+	// +optional
+	EnableDeveloperMode *bool `json:"enableDeveloperMode,omitempty"`
+}
+
+// StorageOptions describes options of storage.
+type StorageOptions struct {
+	// metadata controls shared metadata for the volume claim for this rack.
+	// At this point, the values are applied only for the initial claim and are not reconciled during its lifetime.
+	// Note that this may get fixed in the future and this behaviour shouldn't be relied on in any way.
+	// +optional
+	Metadata ObjectTemplateMetadata `json:"metadata,omitempty"`
+
+	// capacity describes the requested size of each persistent volume.
+	Capacity string `json:"capacity"`
+
+	// storageClassName specifies the name of a storageClass to request.
+	// +optional
+	StorageClassName *string `json:"storageClassName,omitempty"`
+}
+
+type TLSCertificateType string
+
+const (
+	TLSCertificateTypeOperatorManaged TLSCertificateType = "OperatorManaged"
+	TLSCertificateTypeUserManaged     TLSCertificateType = "UserManaged"
+)
+
+type UserManagedTLSCertificateOptions struct {
+	// secretName references a kubernetes.io/tls type secret containing the TLS cert and key.
+	SecretName string `json:"secretName"`
+}
+
+type OperatorManagedTLSCertificateOptions struct {
+	// additionalDNSNames represents external DNS names that the certificates should be signed for.
+	// +optional
+	AdditionalDNSNames []string `json:"additionalDNSNames,omitempty"`
+
+	// additionalIPAddresses represents external IP addresses that the certificates should be signed for.
+	// +optional
+	AdditionalIPAddresses []string `json:"additionalIPAddresses,omitempty"`
+}
+
+type TLSCertificate struct {
+	// type determines the source of this certificate.
+	Type TLSCertificateType `json:"type"`
+
+	// userManagedOptions specifies options for certificates manged by users.
+	// +optional
+	UserManagedOptions *UserManagedTLSCertificateOptions `json:"userManagedOptions,omitempty"`
+
+	// operatorManagedOptions specifies options for certificates manged by the operator.
+	// +optional
+	OperatorManagedOptions *OperatorManagedTLSCertificateOptions `json:"operatorManagedOptions,omitempty"`
+}
+
+// AlternatorOptions holds Alternator settings.
+type AlternatorOptions struct {
+	// writeIsolation specifies the isolation level.
+	WriteIsolation string `json:"writeIsolation,omitempty"`
+
+	// servingCertificate references a TLS certificate for serving secure traffic.
+	// +kubebuilder:default:={type:"OperatorManaged"}
```

Comment:
let's drop these in favour of webhook validation (globally)
We have agreed to proceed with webhooks for the future and mixing the 2 kind of sucks.

Also this validation style can't be unit tested - you'd have to do the testing matrix in e2e.

Reply:
This is default, we don't have mutating webhook so defaulting is done in the OpenAPI schema.

Reply:
ok, lets keep it